### DISCUSSION TITLE
Fireproof To Do List Demo

### DISCUSSION THREAD

**Discussion by jimpick** [modelcontextprotocol/.github#80] [Show and tell]
### Pre-submission Checklist

- [X] This post follows the MCP community guidelines

### What would you like to share?

[Fireproof](https://fireproof.storage/) is an embeddable database designed to bring live data anywhere.

As an teaching example, we started with the [MCP TypeScript Server Tutorial](https://modelcontextprotocol.io/docs/first-server/typescript), and created a simple "To Do List" database.

It was pretty easy, as Fireproof is already embeddable in Node.js and web pages using TypeScript. It has a very simple schema,
so it is ideal to learn from.

![Screenshot of adding a To Do Item](https://raw.githubusercontent.com/jimpick/mcp-fireproof-todos/refs/heads/main/images/butter.png)

I published the MCP server source and a detailed README with a walkthrough here:

* https://github.com/jimpick/mcp-fireproof-todos

Fireproof + MCP is clearly a powerful combo. Some very interesting things could be built with these core technologies.

**Comment by jchris** [comment]
Thanks Jim for showing how simple it is to do this integration!

For our next steps we plan to show how device / cloud sync can bring these datasets anywhere, making it so you can use Fireproof to manage eg, summarizing a bunch of PDFs in a cloud storage bucket, and then make those summaries available to your local Claude.

Another next step that will be fairly simple is making it so the user can switch between profiles, and then eg: have a database that is shared with a team, so everyone's local Claude is using the same data. For instance you could have local Claude write a Christmas list, and then share with Santa, who can have elves (lighter weight models) check off the items as they are built, while live syncing with the local desktop Claude.
